plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1419,3 +1422,249 @@ def synthetic_radiograph(
+            objects defines the composition of the film stack.
+
+        energies : `~astropy.units.Quantity` array, shape [nenergies,]
+            Energies axis over which to calculate the deposition. Units convertible
+            to eV.
+
+        dx : `~astropy.units.Quantity`, optional
+            The spatial resolution of the numerical integration of the stopping power.
+            Defaults to 1 um.
+
+        return_only_active : `bool`, optional
+            If `True`, only the deposition in layers in which the active property
+            is `True` will be returned. This is usually desirable, since particles
+            captured in other layers will not be measured. If `False`, deposition in
+            all layers of the stack are returned. The default is `True`.
+
+        Returns
+        -------
+
+        deposited : `~numpy.ndarray`, shape [nlayers, nenergies]
+            The fraction of an ensemble of each energy that will be deposited in
+            each layer of film. The array is normalized such that the sum
+            along the first dimension (all of the layers) for each population
+            is unity.
+
+        """
+
+        energies = energies.to(u.MeV).value
+
+        # Deposited energy in MeV
+        deposited = np.zeros([len(self._layers), energies.size])
+
+        for i, layer in enumerate(self._layers):
+
+            # Interpolate stopping power for each energy
+            # stopping power here is in MeV/cm
+            sp_fcn = interp1d(
+                layer.energy_axis.to(u.MeV).value,
+                layer.stopping_power.to(u.MeV / u.cm).value,
+                fill_value=(0, np.inf),
+                bounds_error=False,
+            )
+
+            # Slice the layer into sublayer dx thick
+            nsublayers = int(
+                np.floor(layer.thickness.to(u.um).value / dx.to(u.um).value)
+            )
+            sublayers = np.ones(nsublayers) * dx.to(u.um)
+            # Include any remainder in the last sublayer
+            sublayers[-1] += layer.thickness.to(u.um) % dx.to(u.um)
+
+            # Calculate the energy deposited in each sublayer
+            # This is essentially numerically integrating the stopping power
+            for ds in sublayers:
+                # Interpolate the stopping power at the current energies
+                interpolated_stopping_power = sp_fcn(energies)
+                # dE is in MeV
+                dE = interpolated_stopping_power * ds.to(u.cm).value
+
+                # If dE > E for a given energy, set dE=E (stop the particle)
+                dE = np.where(dE > energies, energies, dE)
+
+                energies += -dE
+                deposited[i, :] += dE
+
+        # Normalize the deposited energy array so that each number represents
+        # the fraction of a population of particles of that energy stopped
+        # in that layer.
+        deposited = deposited / np.sum(deposited, axis=0)
```

Comment:
A tiny bit less code 😜 
```suggestion
        deposited /= np.sum(deposited, axis=0)
```